Tools/jit/build.py

```diff
@@ -0,0 +1,695 @@
+"""A template JIT for CPython 3.13, based on copy-and-patch."""
+
+# XXX: I'll probably refactor this file a bit before merging to make it easier
+# to understand... it's changed a lot over the last few months, and the way
+# things are done here definitely has some historical cruft. The high-level
+# architecture and the actual generated code probably won't change, though.
+
+import argparse
+import asyncio
+import dataclasses
+import enum
+import hashlib
+import json
+import os
+import pathlib
+import re
+import shlex
+import subprocess
+import sys
+import tempfile
+import typing
+
+import llvm
+import schema
+
+if sys.version_info < (3, 11):
+    raise RuntimeError("Building the JIT compiler requires Python 3.11 or newer!")
+
+TOOLS_JIT_BUILD = pathlib.Path(__file__).resolve()
+TOOLS_JIT = TOOLS_JIT_BUILD.parent
+TOOLS = TOOLS_JIT.parent
+CPYTHON = TOOLS.parent
+INCLUDE = CPYTHON / "Include"
+INCLUDE_INTERNAL = INCLUDE / "internal"
+PYTHON = CPYTHON / "Python"
+PYTHON_EXECUTOR_CASES_C_H = PYTHON / "executor_cases.c.h"
+TOOLS_JIT_TEMPLATE_C = TOOLS_JIT / "template.c"
+
+
+@enum.unique
+class HoleValue(enum.Enum):
+    CODE = enum.auto()
+    CONTINUE = enum.auto()
+    DATA = enum.auto()
+    EXECUTOR = enum.auto()
+    GOT = enum.auto()
+    OPARG = enum.auto()
+    OPERAND = enum.auto()
+    TARGET = enum.auto()
+    TOP = enum.auto()
+    ZERO = enum.auto()
+
+
+@dataclasses.dataclass
+class Hole:
+    offset: int
+    kind: schema.HoleKind
+    value: HoleValue
+    symbol: str | None
+    addend: int
+    replace = dataclasses.replace
+
+
+S = typing.TypeVar("S", schema.COFFSection, schema.ELFSection, schema.MachOSection)
+R = typing.TypeVar(
```

Comment:
Most likely you mean `_R = TypeVar("_R", bound=schema.COFFRelocation | schema.ELFRelocation | schema.MachORelocation)`

Using type var values seems not needed here.

Reply:
Are you sure? All uses of these expect one of three distinct types, never a union.